src/NLog/Internal/XmlHelper.cs

```diff
@@ -58,7 +59,33 @@ public static class XmlHelper
         /// </summary>
         private static string RemoveInvalidXmlChars(string text)
         {
-            return String.IsNullOrEmpty(text) ? "" : InvalidXmlChars.Replace(text, "");
+            if (string.IsNullOrEmpty(text))
+                return string.Empty;
+
+#if NET3_5
+            return InvalidXmlChars.Replace(text, "");
+#else
+            StringBuilder sb = null;
+            for (int i = 0; i < text.Length; ++i)
+            {
+                char ch = text[i];
+                if (XmlConvert.IsXmlChar(ch))
+                {
+                    if (sb != null)
+                        sb.Append(ch);
```

Comment:
But then I have to allocate a charArray. Anyway no big issue the chance of meeting the forbidden chars are super low. So it will only be a in rare cases. SubString is fine by me.